launcher/src/main/java/org/wildfly/core/launcher/StandaloneCommandBuilder.java

```diff
@@ -102,29 +124,33 @@ public static StandaloneCommandBuilder of(final String wildflyHome) {
      */
     public StandaloneCommandBuilder addJavaOption(final String jvmArg) {
         if (jvmArg != null && !jvmArg.trim().isEmpty()) {
-            final Argument argument = Arguments.parse(jvmArg);
-            switch (argument.getKey()) {
-                case SERVER_BASE_DIR:
-                    if (argument.getValue() != null) {
-                        setBaseDirectory(argument.getValue());
-                    }
-                    break;
-                case SERVER_CONFIG_DIR:
-                    if (argument.getValue() != null) {
-                        setConfigurationDirectory(argument.getValue());
-                    }
-                    break;
-                case SERVER_LOG_DIR:
-                    if (argument.getValue() != null) {
-                        setLogDirectory(argument.getValue());
-                    }
-                    break;
-                case SECURITY_MANAGER_PROP:
-                    setUseSecurityManager(true);
-                    break;
-                default:
-                    javaOpts.add(argument);
-                    break;
+            if (jvmArg.startsWith("-javaagent:")) {
+                agents.add(jvmArg.substring(11));
```

Comment:
Maybe we shouldn't remove the `-javaagent:` here and add it in the `addAgent()` method instead.